static/app/views/settings/projectIssueGrouping/upgradeGrouping.tsx

```diff
@@ -67,38 +102,35 @@ function UpgradeGrouping({
     } catch {
       handleXhrErrorResponse(t('Unable to upgrade config'));
     }
-  };
+  }
 
-  if (!groupingConfigs) {
-    return null;
+  function openUpgradeGroupingConfirmModal() {
+    if (location.hash !== `#${upgradeGroupingId}` || noUpdates || !groupingConfigs) {
```

Comment:
Why not just explicitly check this in the useEffect` and have a single function that handles the `openConfirmModal`?

Reply:
hm I remember we had a lint rule that yelled at you if you used conditions inside of `componentDidMount`, so I was trying to keep everything out of `useEffect` too but I can do that

Reply:
I would think of `useEffect` as a pretty different thing that `componentDidMount`. For one thing, you can have multiple of them, so when you encapsulate logic, you can have it so you have one specific useEffect that is responsabile for that thing.

I also think it's good to keep the dependencies clearly inside of the `useEffect`.

Reply:
I would say it depends... `useEffect` replaces `componentDidMount` and you can use multiple logics inside one `useEffect`

Reply:
>  // Similar to componentDidMount and componentDidUpdate:
>   useEffect(() => {
>     // Update the document title using the browser API
>     document.title = `You clicked ${count} times`;
>   });